docs/blog/posts/2025-04-14-incident-report-organization-team-privileges.md

```diff
@@ -0,0 +1,223 @@
+---
+title: "Incident Report: Organizations Team privileges"
+description: We responded to an incident related to privileges persisting
+    via Organization Teams after Members are removed from Organizations.
+authors:
+  - ewdurbin
+date: 2025-04-14
+tags:
+  - transparency
+  - security
+---
+
+On April 14, 2025 <security@pypi.org> was notified of a potential security concern
+relating to privileges granted to a PyPI User via Organization Teams membership
+persisting after the User was removed from the PyPI Organization the Team belongs to.
+
+We validated the report as a true finding, identified all cases where this scenario
+had occurred, notified impacted parties, and released a fix.
+A full audit determined that all instances were accounted for,
+with no unauthorized actions taken as a result of the issue.
+
+<!-- more -->
+
+## Timeline of events
+
+- 2025-04-14 16:37 UTC
+    A PyPI User who has been testing out our Organizations features noticed the issue
+    and reported it according to our [Security Policy](https://pypi.org/security/)
+    to <security@pypi.org>.
+- 2025-04-14 17:02 UTC
+    PyPI Security acknowledges reciept.
+- 2025-04-14 17:22 UTC
+    PyPI Security validates the report as a true finding.
+- 2025-04-14 17:58 UTC
+    PyPI Security validating test and hot fix prepared for internal review.
+- 2025-04-14 18:30 UTC
+    PyPI Security removes invalid Team Membership and notifies the owners of the only
+    other actively impacted Organization.
+    @EWDurbin opens [public PR](https://github.com/pypi/warehouse/pull/17957) with fix.
+- 2025-04-14 18:33 UTC
+    Hot fix is merged.
+- 2025-04-14 18:39 UTC
+    Hot fix deployed and live on PyPI.
+- 2025-04-14 19:06 UTC
+    Complete security audit complete, validating that only two instances of this had
+    occurred, with no unauthorized actions taken as a result of the persisted
+    privileges.
+
+## Details
+
+PyPI Organizations have been a feature on PyPI since they were first enabled
+on April 20, 2023.
```

Comment:
March*

Reply:
I def flipped the switch on 4202023 at pycon lol.

Reply:
sorry sorry I didn't read the year!